src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2020 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.utils.properties;
+
+import com.epam.ta.reportportal.ws.model.attribute.ItemAttributesRQ;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * Util for retrieving properties from `System` env variables and provided `resource` and converting them to the {@link ItemAttributesRQ}
+ * with {@link ItemAttributesRQ#isSystem()}='true'
+ *
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class SystemAttributesExtractor {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(SystemAttributesExtractor.class);
+
+	private static final String ATTRIBUTE_VALUE_SEPARATOR = "|";
+
+	private SystemAttributesExtractor() {
+		//static only
+	}
+
+	/**
+	 * Loads properties from the specified location
+	 *
+	 * @param resource Path to the resource in classpath
+	 * @return {@link Set} of {@link ItemAttributesRQ}
+	 */
+	public static Set<ItemAttributesRQ> extract(final String resource) {
+		Properties properties = new Properties();
+
+		ofNullable(resource).flatMap(res -> ofNullable(SystemAttributesExtractor.class.getClassLoader().getResource(res)))
+				.ifPresent(url -> {
+					try (InputStreamReader inputStreamReader = new InputStreamReader(url.openStream(), StandardCharsets.UTF_8)) {
```

Comment:
can be moved to the separate method with input stream and property as parameters

Reply:
Stream opening requires IOException handling, so it won't be less complicated after moving in the separate method